File: frontend/src/types/viagem.ts
import { z } from "zod";

export const RotaSchema = z.object({
  id: z.string(),
  origem: z.string(),
  destino: z.string(),
  duracaoEstimada: z.string(),
});

export const ViagemSchema = z.object({
  id: z.string(),
  rotaId: z.string(),
  dataPartida: z.iso.datetime(),
  preco: z.number().positive(),
  capacidade: z.number().int().positive(),
  assentosDisponiveis: z.number().int().nonnegative(),
});

export const ViagemDetalhadaSchema = ViagemSchema.extend({
  rota: RotaSchema,
});

export type Rota = z.infer<typeof RotaSchema>;
export type Viagem = z.infer<typeof ViagemSchema>;
export type ViagemDetalhada = z.infer<typeof ViagemDetalhadaSchema>;

File: backend/OnibusExpress.Api/Models/Viagem.cs
namespace OnibusExpress.Api.Models;

public class Viagem
{
    public string Id { get; set; } = string.Empty;
    public string RotaId { get; set; } = string.Empty;
    public DateTime Data { get; set; }
    public decimal Preco { get; set; }
    public int Capacidade { get; set; }
    public int AssentosDisponiveis { get; set; }
}

public class ViagemDetalhada : Viagem
{
    public Rota Rota { get; set; } = new();
}
